### PR TITLE
Remove Rosetta GitHub workflow

### LINKED ISSUE #7109: Rosestta validation workflow jobs fail due to no account balance files
### Description

Since testnet has disabled account balance files, the rosetta validations jobs in the github workflow fails with a partial mirrornode setup and no true source account balance info. 

### Steps to reproduce

check the workflow runs result

### Additional context

_No response_

### Hedera network

testnet

### Version

v0.91.0-rc1

### Operating system

None

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

Removes the Rosetta GitHub workflow since all the validations fail due to lack of account balance files in testnet.

* Bump `securego/gosec` to `v2.18.2`
* Move gosec job to the security workflow
* Remove Rosetta GitHub workflow

**Related issue(s)**:

Fixes #7109 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/rosetta.yml
Comment (steven-sheehy) [inline .github/workflows/rosetta.yml:1]: Should we keep the workflow but just disable it? Or will the future FST based workflow be completely different?

Comment (xin-hedera) [reply .github/workflows/rosetta.yml:1]: either is fine for me. If later we can reuse most part of it, we can go back to release 0.91 branch for the removed file

Comment (steven-sheehy) [reply .github/workflows/rosetta.yml:1]: Ok, let's keep as is.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by jascks — APPROVED** [review]
LGTM

**Pull request merged**